### DISCUSSION TITLE
wiki

### DISCUSSION THREAD

**Discussion by kianzarrin** [CitiesSkylinesMods/TMPE#1243] [General]

**Comment by kianzarrin** [comment]
![image](https://user-images.githubusercontent.com/26344691/146825397-4b9ce77b-76ab-460e-b18d-57fdcd24d38e.png)

@aubergine10 don't you think using the name transition node in the wiki can be confusing?

in the code it is called blend node (junction with 2 segments). but that sounds like too low level.
transition flag is a node that contains segments with different levels.

**Reply by originalfoo** [reply]
What is meant by `different levels` (eg. road category, or elevations, or...)?

**Reply by kianzarrin** [reply]
small/meduim/large/highway

**Reply by kianzarrin** [reply]
Transition flag is mainly used by highway roads to show highway sign at urban/highway intersections.

**Reply by originalfoo** [reply]
I think the term `Transition node` was derived from the transition flag kinda thing, ie, the type of node (from user perspective) that occurs when roads from different level meet (eg. small road transitions to a medium road).

Do those `gaps with 2 crosswalks` appear anywhere else (ie. anywhere that the transition flag isn't set)?

**Reply by kianzarrin** [reply]
if the pedestrian width or road width do not match, its a blend junction. so between vanilla large road with parking and vanilla large road with trees for-example.

**Comment by krzychu124** [comment]
@aubergine10 Also related to wiki. Would be nice to include @DaEgi01 mod [Check Road Access for Growables](https://steamcommunity.com/sharedfiles/filedetails/?id=2454302667) as a great helper for searching for the cause of service vehicle spawning problems (flickering 0/1/0). 
It can be combined with `BrokenNodeDetector` experimental "Search for disconnected buildings" feature. `BND` can detect disconnected buildings marked by `Check Road Access for Growables` (and also others) and cycle through like when searching for broken public transport stops or broken nodes. 
Sometimes just rotated building with Move It gets disconnected. Egi's mod works great when disconnected buildings don't have other problems but fails if notifications with higher priority are present -> `no road connection` notification has lower priority than e.g. `no water`, `dead`, `garbage` etc.

**Comment by originalfoo** [comment]
@krzychu124 Yup, I'll add that. 👍 

BTW, new image for BND showing latest options...
![broken-nodes-detector](https://user-images.githubusercontent.com/1386719/148626220-6591e761-58b8-4077-be34-719d23e9bd37.jpg)

**Reply by originalfoo** [reply]
Wiki article for flickering 0/1 issue : https://github.com/CitiesSkylinesMods/TMPE/wiki/Vehicle-Count-Flickering-Between-1-and-0

**Reply by originalfoo** [reply]
Looks like the disconnected building detector works! 
![image](https://user-images.githubusercontent.com/1386719/148628007-ffe05555-ad09-41e9-bdeb-b09674a3d403.png)

**Comment by kianzarrin** [comment]
@aubergine10  you should stop before zerbra crossings if traffic is blocked:
![image](https://user-images.githubusercontent.com/26344691/165484919-4b6c5c62-645f-4e40-a8ff-34384c52cfaa.png)
so enter blocked junctions can be modified on transitions.

**Comment by kianzarrin** [comment]
@aubergine10 
> **Transition nodes** (sometimes called **Stretch nodes**)

Stretch node is a NC only node. TMPE does not know about it. it only works where there are only two segments and tries to mimic bend node in functionality (no u-turns though!). There for I deleted those parentheses.

Transition node is also confusing because node gets transition flag on urban-highway junctions.

**Reply by kianzarrin** [reply]
I made it like this 
![image](https://user-images.githubusercontent.com/26344691/165487489-47826cf0-cb0f-4d91-82bd-21e472755a33.png)
